GitCommands/Git/GitModule.cs

```diff
@@ -1812,6 +1828,52 @@ public string UnstageFiles(IReadOnlyList<GitItemStatus> files)
             return output.ToString();
         }
 
+        /// <summary>
+        /// Batch unstage files using <see cref="ExecutableExtensions.RunBatchCommand(IExecutable, ICollection{BatchArgumentItem}, Action{BatchProgressEventArgs}, byte[], bool)"/>
+        /// </summary>
+        /// <param name="selectedItems">Selected file items</param>
+        /// <param name="action">Progress update callback</param>
+        /// <returns><see langword="true" /> if changes should be rescanned; otherwise <see langword="false" /></returns>.
+        public bool BatchUnstageFiles(IEnumerable<GitItemStatus> selectedItems, Action<BatchProgressEventArgs> action = null)
+        {
+            var files = new List<GitItemStatus>();
+            var filesToRemove = new List<string>();
+            var shouldRescanChanges = false;
+            foreach (var item in selectedItems)
+            {
+                if (!item.IsNew)
+                {
+                    filesToRemove.Add(item.Name);
+
+                    if (item.IsRenamed)
+                    {
+                        filesToRemove.Add(item.OldName);
+                    }
+
+                    if (item.IsDeleted)
+                    {
+                        shouldRescanChanges = true;
+                    }
+                }
+                else
+                {
+                    files.Add(item);
+                }
+            }
+
+            if (filesToRemove.Count > 0)
+            {
+                var args = GitCommandHelpers.ResetCmd(ResetMode.ResetIndex, "HEAD");
+                _gitExecutable.RunBatchCommand(new ArgumentBuilder() { args }
+                    .BuildBatchArguments(filesToRemove),
+                    action);
+            }
+
+            UnstageFiles(files);
+
+            return shouldRescanChanges;
+        }
```

Comment:
Can we add tests for this?

Reply:
This one is tricky because it calls UnstageFiles internally. Had to fix NRE for MockProcess.Verify. Test cases added.